app/features/routes.py

```diff
@@ -0,0 +1,75 @@
+from app import db
+from app.features import bp
+from flask import render_template, request
+from flask_login import login_required
+from app.dummy_feature.models import Sensor, sensorFeature, SensorType, TemperatureData, Feature, Room
+import random
+import json
+import datetime
+
+
+# Feature selection page
+@bp.route('/feature_select')
+@login_required
+def feature_selector():
+    features = Feature.query.all()
+    if len(features) <= 0:
+        sensor = Sensor.query.get(1)
+        feature = Feature(name='Temperature Pot de fleur', enum_type='temperature')
+        feature.sensors.append(sensor)
+        db.session.add(feature)
+        db.session.commit()
+        features = Feature.query.all()
+    return render_template('features/feature_select.html', features=features)
+
+
+# Temperature feature page
+@bp.route('/temperature')
+@login_required
+def temperature():
+    feature_one = Feature.query.get(1)
+    sensor_number = request.args.get('sensor', default=None, type=int)
+    sensor = None
+    temp = None
+    time = None
+    sensors = Sensor.query.all()
+    if sensor_number is not None:
+        sensor = Sensor.query.get(sensor_number)
+    if sensor is not None:
+        temperature_plus = TemperatureData.query.filter_by(sensor_id=sensor.id).all()
+        temp = []
+        time = []
+        date_format = '%Y-%m-%d %H:%M:%S'
+        for a in temperature_plus:
+            # a_temp = {'value': a.value, 'snapshot_date': a.snapshot_date}
+            temp.append(a.value)
+            time.append(a.snapshot_date.strftime(date_format))
```

Comment:
Loop “for” can crash if “temperature_plus” is null, check if this variable is not null and do the loop.